src/Build/Evaluation/LazyItemEvaluator.LazyItemOperation.cs

```diff
@@ -342,6 +346,14 @@ protected static bool ItemspecContainsASingleItemReference(ItemSpec<P, I> itemSp
                     return false;
                 }
 
+                // If the itemSpec is a single call to an item function, like @(X->Something(...)), it may get this
```

Comment:
I guess only item function transforms are impacted by this since I can't do `@(X-> '%(Identity)')` in lazy evaluations?

Reply:
That's still an item transform. It should be supported in lazy evaluations.
The safest is for this function to reject all transforms, since there's no guarantee the transform would produce the same result as the original item.